src/backend/routers/agent.py

```diff
@@ -98,6 +109,30 @@ async def create_agent(
                 await update_or_create_tool_metadata(
                     created_agent, tool_metadata, session, ctx
                 )
+
+        # Consolidate agent files into one index in compass
```

Comment:
Upload and agent creation are two steps. 

Upload indexes agent files into individual orphaned indices. When an agent is being created with those file IDs as the tools metadata we consolidate them under one index (agent ID) here.